system/firewalld.py

```diff
@@ -112,13 +112,197 @@
 # Globals
 #
 fw = None
+msgs = None
 module = None
-fw_offline = False
+changed = False
+immediate = None
+permanent = None
 Rich_Rule = None
+fw_offline = False
+desired_state = None
 FirewallClientZoneSettings = None
 
 module = None
 
+
+#####################
+# action logic
+#
+def action_runner(
+    action,
+    get_enabled_immediate, get_enabled_immediate_args,
+    get_enabled_permanent, get_enabled_permanent_args,
+    set_enabled_immediate, set_enabled_immediate_args,
+    set_enabled_permanent, set_enabled_permanent_args,
+    set_disabled_immediate, set_disabled_immediate_args,
+    set_disabled_permanent, set_disabled_permanent_args
```

Comment:
I'd be tempted to use functools.partial here if you already require python-2.6+. https://docs.python.org/2/library/functools.html#functools.partial  functools was added in python2.5 so you can't use it if you're stuck on python-2.4.

I do see that you use set_enabled_permanent_args and set_enabled_immediate_args to construct the enabled_msg and disabled_msg... I guess if you use partial the msg strings will have to be constructed outside and then passed into the function.